src/modalities/models/model_factory.py

```diff
@@ -132,3 +245,83 @@ def get_activation_checkpointed_model(model: FSDP, activation_checkpointing_modu
                     f"Current model type: {type(model)}"
                 )
         return model
+
+    @staticmethod
+    def get_compiled_model(model: nn.Module, block_names: list[str]) -> nn.Module:
+        """
+        Apply torch.compile to each transformer block, which makes compilation efficient due to
+        repeated structure. Alternatively one can compile the whole model (after applying DP).
+        Inspired by: https://github.com/pytorch/torchtitan/blob/6b2912a9b53464bfef744e62100716271b2b248f/torchtitan/parallelisms/parallelize_llama.py#L275
+        """
+
+        def get_parent_module_and_child_name(child_module: nn.Module, model: nn.Module) -> tuple[nn.Module, str]:
```

Comment:
It does not return child_name. Please rename function.

Reply:
This seems already correct to me. Was this corrected maybe already in the mean-time?